src/commonMain/kotlin/fr/acinq/lightning/channel/states/Channel.kt

```diff
@@ -161,7 +161,75 @@ sealed class ChannelState {
         ChannelAction.Blockchain.SendWatch(WatchConfirmed(channelId, tx.tx, staticParams.nodeParams.minDepthBlocks.toLong(), BITCOIN_TX_CONFIRMED(tx.tx)))
     )
 
-    abstract fun ChannelContext.handleLocalError(cmd: ChannelCommand, t: Throwable): Pair<ChannelState, List<ChannelAction>>
+    internal fun ChannelContext.handleLocalError(cmd: ChannelCommand, t: Throwable): Pair<ChannelState, List<ChannelAction>> {
+        logger.error(t) { "error on command ${cmd::class.simpleName}" }
+
+        fun abort(channelId: ByteVector32?): Pair<ChannelState, List<ChannelAction>> {
+            val error = channelId?.let { Error(it, t.message) }
+            val actions = error?.let { listOf(ChannelAction.Message.Send(it)) } ?: emptyList()
+            return Pair(Aborted, actions)
+        }
+
+        fun forceClose(state: ChannelStateWithCommitments): Pair<ChannelState, List<ChannelAction>> {
+            val error = Error(state.channelId, t.message)
+            return when {
+                state.commitments.nothingAtStake() -> Pair(Aborted, listOf(ChannelAction.Message.Send(error)))
```

Comment:
This is not marking the channel as `Closed` in the DB, which is probably an issue...I will rebase #478 and use the `RemoveChannel` action in that case.

Reply:
Done as part of the rebase.

Reply:
I went ahead and completely removed the `nothingAtStake` case in [b6d7395](https://github.com/ACINQ/lightning-kmp/pull/479/commits/b6d73953f9514cd8486a7ba525e7f25a16ca2f55).